pkg/controller/controller_utils.go

```diff
@@ -284,7 +284,8 @@ func filterActivePods(pods []api.Pod) []*api.Pod {
 	var result []*api.Pod
 	for i := range pods {
 		if api.PodSucceeded != pods[i].Status.Phase &&
-			api.PodFailed != pods[i].Status.Phase {
+			api.PodFailed != pods[i].Status.Phase &&
+			pods[i].DeletionTimestamp != nil {
```

Comment:
FYI, this will make kubectl stop return as soon as all the pods have been marked for deletion. Some may still be running. I'm ok with that, but we should probably document it.